builtin/stash--helper.c

```diff
@@ -634,6 +640,55 @@ static int list_stash(int argc, const char **argv, const char *prefix)
 	return ret;
 }
 
+static int show_stash(int argc, const char **argv, const char *prefix)
+{
+	int i, ret = 0;
+	struct argv_array args = ARGV_ARRAY_INIT;
+	struct argv_array args_refs = ARGV_ARRAY_INIT;
+	struct stash_info info;
+	struct option options[] = {
+		OPT_END()
+	};
+
+	argc = parse_options(argc, argv, prefix, options,
+			git_stash_helper_show_usage, PARSE_OPT_KEEP_UNKNOWN);
+
+	argv_array_push(&args, "diff");
+
+	for (i = 0; i < argc; ++i) {
+		if (argv[i][0] == '-')
+			argv_array_push(&args, argv[i]);
+		else
+			argv_array_push(&args_refs, argv[i]);
+	}
+
+	ret = get_stash_info(&info, args_refs.argc, args_refs.argv);
```

Comment:
If there is no stash, I think we should `die()`, just like `assert_stash_like()` does. That is not, however, what `get_stash_info()` does, so we have to do it here, manually, based on the return value.

Reply:
I think that a better solution for this is to edit `assert_stash_like()` into a void function to be more similar with `assert` from `<assert.h>`. Since `struct stash_info *info` is given as parameter, it can be freed inside the `assert_stash_*()` functions.